lib/server-builder.js

```diff
@@ -107,22 +100,17 @@ class ServerBuilder {
   updateDependencyLines (configurationContent, additionalAppDependencies) {
     const dependencyLines = additionalAppDependencies
       .map(function (dependency) {
-        if (/[\s']/.test(dependency)) {
-          throw new Error(`Single quotes and space characters are disallowed in additional dependencies: ${dependency}`);
+        if (/[\s'\\$]/.test(dependency)) {
+          throw new Error(`Single quotes, dollar characters and whitespace characters are disallowed in additional dependencies: ${dependency}`);
         }
         return `implementation '${dependency}'`;
       })
-      .join('\n');
-
-    return configurationContent.split(/\n/)
-      .map(function (line) {
-        if (line.trim() === GRADLE_DEPENDENCIES_PLACEHOLDER) {
-          return dependencyLines;
-        } else {
-          return line;
-        }
-      })
-      .join('\n');
+      .join('$1');
```

Comment:
will it work this way?

Reply:
Yes, this uses new line characters from the capture group in the pattern. I think I was wrong that I didn't add a comment here.